include/openthread/platform/border_routing.h

```diff
@@ -65,6 +65,22 @@ extern "C" {
  */
 extern void otPlatBorderRoutingProcessIcmp6Ra(otInstance *aInstance, const uint8_t *aMessage, uint16_t aLength);
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE
+#if !OPENTHREAD_BORDER_ROUTING_DHCP6_PD_REQUESTOR
```

Comment:
We should not add any `#if` in any of public/platform headers (since they may be included in other projects/libraries and wont see the same set of configs as OT core stack).

We instead document that the public API is available or platform API is used when certain `OPENTHREAD_CONFIG` is enabled